src/cassandra/cluster.rs

```diff
@@ -157,11 +156,12 @@ impl Cluster {
     }
 
     /// Performs a blocking call to connect to Cassandra cluster
-    pub fn connect(&mut self) -> Result<Session> {
+    pub async fn connect(&mut self) -> Result<Session> {
```

Comment:
This is a breaking change - as to avoid having to pick a runtime to drive the future to completion, leaving that as an exercise for the caller (e.g. using tokio's runtime block-on, or just awaiting it...)